data/test/scenarios/test_remove_object_compatibility.cfg

```diff
@@ -0,0 +1,118 @@
+# This unit test is used for verify backward compatibility of remove_ability
+
+{GENERIC_UNIT_TEST "test_remove_object_compatibility" (
+    [event]
+        name=start
+        [modify_unit]
+            [filter]
+            [/filter]
+            max_hitpoints=100
+            hitpoints=100
+            attacks_left=1
+        [/modify_unit]
+        [object]
+            silent=yes
+            [effect]
+                apply_to=new_ability
+                [abilities]
+                    [chance_to_hit]
+                        id=test_cth_0
+                        value=0
+                    [/chance_to_hit]
+                    [chance_to_hit]
+                        id=test_cth
+                        value=100
+                    [/chance_to_hit]
+                [/abilities]
+            [/effect]
+            [filter]
+                id=bob
+            [/filter]
+        [/object]
+        [object]
+            silent=yes
+            [effect]
+                apply_to=new_ability
+                [abilities]
+                    [chance_to_hit]
+                        id=test_cth_0
+                        value=0
+                    [/chance_to_hit]
+                    [chance_to_hit]
+                        id=test_cth
+                        value=100
+                    [/chance_to_hit]
+                [/abilities]
+            [/effect]
+            [filter]
+                id=alice
+            [/filter]
+        [/object]
+        [object]
+            [filter]
+                id=bob
+            [/filter]
+            silent=yes
+            [effect]
+                apply_to=remove_ability
+                [abilities]
+                    [chance_to_hit]
+                        id=test_cth
+                    [/chance_to_hit]
```

Comment:
The thing that this is supposed to test is that the tag name here is ignored. So, change it to something other than `chance_to_hit` (but the ability itself still uses `chance_to_hit`). Then try running the unit test on unmodified master to make sure it passes there. If it passes on master, then check if it passes on this branch.

Reply:
In fact tag name no longer ignored, i will replace (return false) in checking by deprecated message.

Reply:
But tag name _is_ ignored on master, correct?

Reply:
correct, a tag is necessary, but tag name can be diffrent of ability tag name, and it is a problem in my point of view. the logic would what wml devellopper use the right tag name but if this particularity is know, impose the checking without advertissement will cause problem